src/packet_macros.rs

```diff
@@ -29,7 +29,7 @@ macro_rules! packet_boilerplate {
             match packet {
                 $(Packet::$name(packet) => {
                     write_var_int(&mut cursor, $name::ID);
-                    packet.write(&mut cursor)
+                    packet.write_fields(&mut cursor)
```

Comment:
Changed this name to be more accurate- since this method doesn't write any of the headers of the packet